### PR TITLE
Add filter category on portal profile

### PR DESCRIPTION
**Pull request by lzammit**
## Description

Now you can create a portal profile based the category (role) of a node.
## NEWS file entries

Add category filter on portal profile

Enhancements
++++++++++++
Add category filter on portal profile
## Delete branch after merge

No

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/I18N/i_default.po [hunk 1]
```diff
@@ -5931,6 +5931,10 @@ msgid "pending"
 msgstr "pending"
 
 # pf::filter (Portal Profile Filters)
+msgid "pf::profile::filter::category"
+msgstr "Category - Role"
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/I18N/i_default.po]: I think this should be
"Node category/role" or just 
"Node role"

**Comment by cgx** [comment]
Could we display a select input field with values from the `node_category` database table?

**Comment by jrouzierinverse** [comment]
We could but I would want to keep that as a separate pull request.
Since I am planning to rework the display of the filters.
I would prefer to have the mini-framework there instead of doing a one off just for this filter

**Comment by cgx** [comment]
@jrouzierinverse All right.

@lzammit Even though the table field is named `category`, I would tend to use the new nomenclature and name the module `pf::profile::filter::role`.

**Comment by jrouzierinverse** [comment]
Would the name pf::profile::filter::node_role be better?

**Comment by cgx** [comment]
Sure!

**Comment by lzammit** [comment]
done @louismunro

**Comment by lzammit** [comment]
please merge @julsemaan

**Comment by julsemaan** [comment]
Why is there two new filters added. Also you changed the module name (which is good) but you changed the matching attribute at the same time which to my eyes looks like it's going to break the filter. Please fix these

**Comment by lzammit** [comment]
please review @julsemaan

**Pull request merged**